### PR TITLE
Towards #56 - improving model_example_direct_prediction.md

### PR DESCRIPTION
**Pull request by jfdelgad**
Towards #56

Changes proposed in this PR:

-  Changes the comparison example to mimic main2.py
-  Fixes typos
-  Add missing final prediction 1h, 2h...12h in the future

### CODE REVIEW

File: examples/model_example_direct_prediction.md [hunk 1]
```diff
@@ -58,14 +60,23 @@ y_train = full_train_close[max_lag:,:]
...
 y_test = full_test_close[max_lag:,:]
 x_test = np.concatenate((full_test_close[0:-max_lag,:],full_test_volume[0:-max_lag,:]),axis=1)
 
+```
+
+## 3.  Make predictions
+
+
+
+### 3.1  rain model and calculate prediction error in the test set
```
Comment (trentmc) [inline examples/model_example_direct_prediction.md:75]: We can think of nd-to-end READMEs like this one as an "expansion" of main2.md. 
- Therefore major section names and numbers should be the same
- This will help contestants to see what each end-to-end README is doing, wrt main2.md and other end-to-end READMEs.

Therefore we want, like main2.md:
- "3.1 Build a simple AI model"
- "3.2 Run the AI model to make future ETH price predictions"
- "3.3 Calculate NMSE"

Please rearrange your content to fit into this. Shouldn't be that hard. 
- Perhaps make your "3.2 Select the best performant model and make final predictions" to be numbered 3.2.1.
- Then your "3.3 Calculate NMSE" may be trivial - just point to the best test_nmse result from the previous section.

File: examples/model_example_direct_prediction.md [hunk 2]
```diff
@@ -15,7 +15,9 @@ In the console:
 pip install pandas numpy requests
 ```
 
-## 2. Create and open a python file
+## 2. Get data locally
+
+In Python:
```
Comment (trentmc) [inline examples/model_example_direct_prediction.md:20]: "In Python:" -> "In the Python console:"

File: examples/model_example_direct_prediction.md [hunk 3]
```diff
@@ -139,12 +150,33 @@ plt.title("Comparison of different methods for predicting ETH value 1-12 hours a
...
+## 5. Discussion
 
-Furthermore, these methods base their prediction in the previous 12 hours and do not take into consideration trends nor other possible variables such as holidays, seasons etc. all this motivates using models that model time and allow for latten variable modeling such as Hidden Markov Models or Recurrent Neural networks (including LSTMS)
+The algorithms presented are just shown as introductory approaches but are not production ready. For instance, we use the close value as the real eth value and an extra feature including the open price is added, however many other (more informative) variables could be incorporated. Some of the methods proposed need tuning of hyper parameters, such as the regularization parameter C in the SVR, number of estimators in the Random forest method, etc.
```
Comment (trentmc) [inline examples/model_example_direct_prediction.md:180]: For the space taken by the two paragraphs "the algorithms presented" and "Furthermore, ...", they don't tell that much new. Perhaps we can shorten them to e.g. "There are many ways to reduce error further, including: more data, tuning hyperparameters, better feature vectors, better modeling algorithms including time-aware ones (e.g. recurrent neural networks)."

Details: here is why we're ok to remove or shorten some lines...

> The algorithms presented are just shown as introductory approaches but are not production ready. 
This should be obvious, it's presented as an "example". And, it's in a README, after all. And, it's for a competition.

> For instance, we use the close value as the real eth value
Not really helpful.

> However many other (more informative) variables could be incorporated
Out of scope for this README. The top-level README already discusses, especially in the section of main README on various data feeds, and the sub-README dedicated to more data ideas

> Some of the methods proposed need tuning of hyper parameters, such as the regularization parameter C in the SVR, number of estimators in the Random forest method, etc.

This will be obvious to data scientists. But it doesn't hurt to have a shortened version, e.g. what I suggested at the top.

**Review by trentmc — CHANGES_REQUESTED** [review]

File: examples/model_example_direct_prediction.md (hunk 3 above)
Comment (jfdelgad) [reply examples/model_example_direct_prediction.md]: Done

File: examples/model_example_direct_prediction.md (hunk 2 above)
Comment (jfdelgad) [reply examples/model_example_direct_prediction.md:20]: Done

File: examples/model_example_direct_prediction.md [hunk 4]
```diff
@@ -58,14 +60,23 @@ y_train = full_train_close[max_lag:,:]
...
 
+```
+
+## 3.  Make predictions
+
 
+
+### 3.1 Build a simple AI model
+
+In the same Python console:
```
Comment (trentmc) [inline examples/model_example_direct_prediction.md:77]: I just noticed, it has the following section. That's redundant with the inital setup. So, please delete.

```python
# normalized mean-squared-error, consistent with elsewhere in predict-eth
def nmse(y, yhat) -> float:
    range_y = max(y) - min(y)    
    return np.sqrt(np.average(((yhat - y) / range_y) ** 2))
```

File: examples/model_example_direct_prediction.md (hunk 2 above)
Comment (trentmc) [inline examples/model_example_direct_prediction.md:20]: "In Python:" -> "In the same Python console:"

**Review by trentmc — APPROVED** [review]
Looks great in general. A couple small tweaks to go, then let's shipped. Anticipating these changes, I already approved
